### PR TITLE
fix(miners): avoid disabled TLS certificate checks

### PR DESCRIPTION
**Pull request by yyswhsccc**
Clean redo of #7603 after Scott's scope-hygiene feedback.

This branch intentionally keeps the original technical fix small and excludes the unrelated shared CI/baseline/consensus-node edits that caused the previous PR to be closed.

Selector provenance:
- natural_owner: `both_selectors`
- selection_bucket: `both_selectors`
- selected_by_lgbm: `True`
- selected_by_native: `True`
- dispatch_arm: `trained_lgbm_selector`
- selector_run_id: `6ac915456973ebaf`

Scoped files:
- `miners/floppy-miner/tools/relay.py`

Validation:
- `python3 -m py_compile miners/floppy-miner/tools/relay.py -> passed`
- `GitHub Contents API path filter -> source/test files only`

Boundaries:
- No payout amount changes.
- No admin keys or production wallet changes.
- No manual wallet crediting.
- No `node/rustchain_v2_integrated_v2.2.1_rip200.py` or `scripts/baselines/*` maintenance diff.

@Scottcjn this is the clean, scoped redo of the earlier closed PR.

wallet: RTC47bc28896a1a4bf240d1fd780f4559b242bcd945

### CODE REVIEW

File: miners/floppy-miner/tools/relay.py [hunk 1]
```diff
@@ -28,19 +28,18 @@
...
-    """Create SSL context that accepts self-signed certs."""
-    ctx = ssl.create_default_context()
-    ctx.check_hostname = False
-    ctx.verify_mode = ssl.CERT_NONE
-    return ctx
+def create_ssl_context(verify_tls: bool = True):
+    """Create an SSL context, preserving platform verification by default."""
+    if verify_tls:
+        return None
+    return ssl._create_unverified_context()
```
Comment (Vyacheslav-Tomashevskiy) [inline miners/floppy-miner/tools/relay.py:35]: `ssl._create_unverified_context()` is a private (underscore-prefixed) API. It works today, but it is not part of the public contract and could change without notice. A more explicit and stable equivalent that documents intent in-line:
```python
ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE
return ctx
```
Minor robustness nit, not a blocker.

Comment (Vyacheslav-Tomashevskiy) [inline miners/floppy-miner/tools/relay.py:35]: Security-hygiene gap: when the insecure path is taken there is no warning emitted. Disabling TLS verification silently is exactly the failure mode that hides MITM exposure from operators. Recommend printing a one-line stderr warning (mirroring urllib3's `InsecureRequestWarning`) when `verify_tls=False`, e.g. `print('[RELAY] WARNING: TLS certificate verification disabled', file=sys.stderr)` in `main()` when `args.insecure_tls` is set, so the downgrade is visible in logs.

File: miners/floppy-miner/tools/relay.py [hunk 2]
```diff
@@ -87,6 +86,11 @@ def main():
     parser.add_argument("--node", default=DEFAULT_NODE, help="RustChain node URL")
     parser.add_argument("--serial", default=None, help="Serial port (e.g., /dev/ttyUSB0)")
     parser.add_argument("--baud", type=int, default=9600, help="Serial baud rate")
+    parser.add_argument(
```
Comment (Vyacheslav-Tomashevskiy) [inline miners/floppy-miner/tools/relay.py:89]: Test-gap: `tools/` contains only `build_floppy.py` and `relay.py` — there are no tests for either branch of `create_ssl_context` or for `forward_attestation`'s new `verify_tls` plumbing. The PR validation is `py_compile` only. A small unit test asserting `create_ssl_context(True) is None` and `create_ssl_context(False).verify_mode == ssl.CERT_NONE` would lock in the secure default and prevent a future regression that silently flips it back.

**Review by Vyacheslav-Tomashevskiy — COMMENTED** [review]
**Review of `miners/floppy-miner/tools/relay.py` @ `bc9de9a` (COMMENTED).**

Verified the fix against the source at head, not just the diff. The core security problem is genuinely resolved: the previous unconditional `check_hostname=False` + `verify_mode=CERT_NONE` made every relay→node connection MITM-vulnerable by default. Now `create_ssl_context(verify_tls=True)` returns `None`, and `urllib.request.urlopen(req, context=None)` (line 52) falls back to the **default verifying** SSL context — so out-of-the-box behaviour is secure. The `verify_tls` keyword is plumbed correctly end-to-end: `forward_attestation(..., verify_tls=not args.insecure_tls)` (line 116) and defaults to `True` at every layer, so omitting the flag keeps verification on. Code-search confirms `create_ssl_context` has exactly **1 occurrence repo-wide**, so the hardening is complete for this module.

Compat check (positive): the documented use case in `README.md` connects to the public `https://rustchain.org/attest/submit`, which presents a valid certificate — so secure-by-default does **not** break the documented path. Only undocumented local/self-signed nodes need the new `--insecure-tls` opt-in, which is the right tradeoff. `mergeable_state` is `unstable`, but that appears to be unrelated base/CI lag rather than this diff.

A few non-blocking findings below (private API, missing insecure-mode warning, test-gap).

**Review by jaxint — COMMENTED** [review]
## Code Review

### Summary
This PR implements security fixes for miners. The changes look good overall.

### Key Changes
- Proper handling of environment variables and security configurations
- Good error handling and edge case coverage

### Testing Recommendations
- Consider adding unit tests for the new validation logic
- Test with various edge cases (empty strings, null values, etc.)

### Security Considerations
✓ The changes improve security posture by properly validating inputs
✓ TLS certificate verification is properly enforced

**Recommendation:** Approve and merge.

---
*Review submitted via RustChain bounty program #1009*

**Comment by Scottcjn** [comment]
Thanks — the direction (scoped TLS/timeout/env hardening) is good and these don't touch the consensus node, which is what we asked for. Two blockers to merging:

1. **CI is red** — the `test` job is failing on these. Please make sure your change passes the full test suite locally (`python3 -m pytest -q`) before submitting; a red `test` check can't be merged.
2. **Volume** — there are **53 open micro-PRs** from this account right now. That's not reviewable at that granularity. Please **consolidate** related fixes into a handful of coherent PRs (e.g. one "bound subprocess timeouts" PR, one "TLS-verify-by-default" PR), each green.

Closing this batch to keep the queue reviewable — a consolidated, CI-green resubmission is genuinely welcome and will be merged.

— Elyan Labs
